src/Symfony/Component/Message/ContainerHandlerLocator.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Message;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Message\Exception\NoHandlerForMessageException;
+
+/**
+ * @author Miha Vrhovnik <miha.vrhovnik@gmail.com>
+ */
+class ContainerHandlerLocator implements HandlerLocatorInterface
```

Comment:
WDYT about `Factory` instead of `Locator`?

Reply:
A bit late 🙊

Reply:
that implies renaming the interface to `HandlerFactoryInterface` right? Tend to agree we're factorizing here :) but current naming is fine also IMHO.